server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/CompactionCoordinator.java

```diff
@@ -0,0 +1,766 @@
+      if (running.isEmpty()) {
+        LOG.info("No compactions running on Compactors.");
+      } else {
+        LOG.info("Found {} running external compactions", running.size());
+        running.forEach((hp, job) -> {
+          // Find the tserver that has this compaction id
+          boolean matchFound = false;
+
+          // Attempt to find the TServer hosting the tablet based on the metadata table
+          // TODO use #1974 for more efficient metadata reads
+          KeyExtent extent = KeyExtent.fromThrift(job.getExtent());
+          LOG.debug("Getting tablet metadata for extent: {}", extent);
+          TabletMetadata tabletMetadata = getMetadataEntryForExtent(extent);
+
+          if (tabletMetadata != null && tabletMetadata.getExtent().equals(extent)
+              && tabletMetadata.getLocation() != null
+              && tabletMetadata.getLocation().getType() == LocationType.CURRENT) {
+
+            TServerInstance tsi = tservers.stream()
+                .filter(
+                    t -> t.getHostAndPort().equals(tabletMetadata.getLocation().getHostAndPort()))
+                .findFirst().orElse(null);
+
+            if (null != tsi) {
+              TabletClientService.Client client = null;
+              try {
+                LOG.debug(
+                    "Checking to see if tserver {} is running external compaction for extent: {}",
+                    tsi.getHostAndPort(), extent);
+                client = getTabletServerConnection(tsi);
+                boolean tserverMatch = client.isRunningExternalCompaction(TraceUtil.traceInfo(),
+                    getContext().rpcCreds(), job.getExternalCompactionId(), job.getExtent());
+                if (tserverMatch) {
+                  LOG.debug(
+                      "Tablet server {} is running external compaction for extent: {}, adding to running list",
+                      tsi.getHostAndPort(), extent);
+                  RUNNING.put(ExternalCompactionId.of(job.getExternalCompactionId()),
+                      new RunningCompaction(job, ExternalCompactionUtil.getHostPortString(hp),
+                          tsi));
+                  matchFound = true;
+                } else {
+                  LOG.debug("Tablet server {} is NOT running external compaction for extent: {}",
+                      tsi.getHostAndPort(), extent);
+                }
+              } catch (TException e) {
+                LOG.warn("Failed to notify tserver {}",
+                    tabletMetadata.getLocation().getHostAndPort(), e);
+              } finally {
+                ThriftUtil.returnClient(client);
+              }
+            } else {
+              LOG.info("Tablet server {} is not currently in live tserver set",
+                  tabletMetadata.getLocation().getHostAndPort());
+            }
+          } else {
+            LOG.info("No current location for extent: {}", extent);
+          }
```

Comment:
I guess the question is whether or not the CompactionCoordinator can get stale data from the metadata table. The code tries to find the TabletServer running this compaction. It's possible that the Tablet could be in the process of being re-hosted (in the case of a TServer crash). In the current implementation an entry would not be made in the internal running list because we didn't find the TServer. If we just use the Metadata entries, we could put incorrect information in the running list.

Reply:
I may be missing something.  I was looking at the code saw a good bit of effort went into finding `tsi`, however it seems after the information was found that it was never used.

Reply:
So, it looks like the CompactionFinalizer scans the Metadata table to determine which TabletServer to notify when a compaction fails or succeeds. We might be able to remove this after all.

Reply:
To be more specific, my IDE indicated nothing calls [RunningCompaction.getTserver()](https://github.com/apache/accumulo/blob/cdfec2afccd22ea344ad8fd3a3ce622ee5b87324/server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/RunningCompaction.java#L58)

Reply:
To be more specific, my IDE indicated nothing calls [RunningCompaction.getTserver()](https://github.com/apache/accumulo/blob/cdfec2afccd22ea344ad8fd3a3ce622ee5b87324/server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/RunningCompaction.java#L58)

Reply:
Agreed, I'm going to push another commit shortly that will significantly reduce the lines of code for the re-start and remove the getTServer method from RunningCompaction.

Reply:
resolved in 90e76cd59e5ed6e08a75020f0d50fe69c4bebee3